code/datums/action.dm

```diff
@@ -673,23 +681,71 @@
 /datum/action/cooldown/IsAvailable()
 	return next_use_time <= world.time
 
-/datum/action/cooldown/proc/StartCooldown()
-	next_use_time = world.time + cooldown_time
-	button.maptext = MAPTEXT("<b>[round(cooldown_time/10, 0.1)]</b>")
+/// Starts a cooldown time, will use default cooldown time if an override is not specified, shares that cooldown with similar abilities
+/datum/action/cooldown/proc/StartCooldown(override_cooldown_time, will_share_cooldowns = TRUE)
+	if(shared_cooldown && will_share_cooldowns)
+		for(var/datum/action/cooldown/C in owner.actions - src)
+			if(shared_cooldown == C.shared_cooldown)
+				if(isnum(override_cooldown_time))
+					C.StartCooldown(override_cooldown_time, FALSE)
+				else
+					C.StartCooldown(cooldown_time, FALSE)
+	if(isnum(override_cooldown_time))
+		next_use_time = world.time + override_cooldown_time
+	else
+		next_use_time = world.time + cooldown_time
 	UpdateButtonIcon()
 	START_PROCESSING(SSfastprocess, src)
 
-/datum/action/cooldown/process()
+/datum/action/cooldown/Trigger(atom/target)
+	. = ..()
+	if(!.)
+		return
 	if(!owner)
+		return FALSE
+	if(click_to_activate)
+		if(target)
+			// For automatic / mob handling
+			return InterceptClickOn(owner, null, target)
+		if(owner.click_intercept == src)
+			owner.click_intercept = null
+			to_chat(owner, "<b>[src] is no longer active</b>")
```

Comment:
```suggestion
			to_chat(owner, "<b>[src] is no longer active.</b>")
```

In an ideal world these are not necessary at all, by the way. The icon itself should have some indication to whether or not its active. If it does, then these do not need to exist.